platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/ThreadSafeConsensusInfo.java

```diff
@@ -0,0 +1,202 @@
+        this.minRound = storage.minIndex();
+
+        if (fameDecidedBelow == ConsensusConstants.ROUND_FIRST) {
+            // if there are no rounds, set the defaults
+            maxRoundGeneration = GraphGenerations.FIRST_GENERATION;
+            minGenNonAncient = GraphGenerations.FIRST_GENERATION;
+            minRoundGeneration = GraphGenerations.FIRST_GENERATION;
+            return;
+        }
+
+        updateMaxRoundGeneration();
+        updateMinGenNonAncient();
+        updateMinRoundGeneration();
+    }
+
+    /**
+     * Update the max round generation
+     *
+     * <p>Executed only on consensus thread.
+     */
+    private void updateMaxRoundGeneration() {
+        final MinGenInfo info = storage.get(getLastRoundDecided());
+        if (info == null) {
+            // this should never happen
+            LOG.error(
+                    LogMarker.EXCEPTION.getMarker(),
+                    "maxRound({}) is null in updateMaxRoundGeneration()",
+                    getLastRoundDecided());
+            return;
+        }
+        long newMaxRoundGeneration = info.minimumGeneration();
+
+        // 13 May 2021
+        // Guarantee that the round generation is non-decreasing. When we have a local events file,
+        // this condition will be implicit, so this clause will be removed.
+        newMaxRoundGeneration = Math.max(newMaxRoundGeneration, maxRoundGeneration);
+
+        maxRoundGeneration = newMaxRoundGeneration;
+    }
+
+    /**
+     * Update the oldest non-ancient round generation
+     *
+     * <p>Executed only on consensus thread.
+     */
+    private void updateMinGenNonAncient() {
+        final long nonAncientRound =
+                RoundCalculationUtils.getOldestNonAncientRound(config.roundsNonAncient(), getLastRoundDecided());
+        final MinGenInfo info = storage.get(nonAncientRound);
+        if (info == null) {
+            // should never happen
+            LOG.error(
+                    LogMarker.EXCEPTION.getMarker(),
+                    "nonAncientRound({}) is null in updateMinGenNonAncient()",
+                    nonAncientRound);
+            return;
+        }
+        minGenNonAncient = info.minimumGeneration();
+    }
+
+    /** Update the min round judge generation. Executed only on consensus thread. */
+    private void updateMinRoundGeneration() {
+        final MinGenInfo info = storage.get(getMinRound());
+        if (info == null) {
+            // this should never happen
+            LOG.error(
+                    LogMarker.EXCEPTION.getMarker(),
+                    "minRound({}) is null in updateMinRoundGeneration()",
+                    getMinRound());
+            return;
+        }
+
+        long newMinRoundGeneration = info.minimumGeneration();
+
+        // 13 May 2021
+        // Guarantee that the round generation is non-decreasing. When we have a local events file,
+        // this condition will be implicit, so this clause will be removed.
```

Comment:
dated comment, says the clause should be removed?

Reply:
updated the comment, still relevant for a short while